stl/inc/xutility

```diff
@@ -864,22 +864,40 @@ _EXPORT_STD template <class _Fn, class... _Its>
     requires (indirectly_readable<_Its> && ...) && invocable<_Fn, iter_reference_t<_Its>...>
 using indirect_result_t = invoke_result_t<_Fn, iter_reference_t<_Its>...>;
 
-_EXPORT_STD template <indirectly_readable _It, indirectly_regular_unary_invocable<_It> _Proj>
-struct projected {
-    using value_type = remove_cvref_t<indirect_result_t<_Proj&, _It>>;
-    indirect_result_t<_Proj&, _It> operator*() const {
-        _CSTD abort();
-    }
+template <class _It>
+struct _Projected_difference_type_impl {
+    struct _Base {};
+};
+
+template <weakly_incrementable _It>
+struct _Projected_difference_type_impl<_It> {
+    struct _Base {
+        using difference_type = iter_difference_t<_It>;
+    };
 };
 
 template <class _It, class _Proj>
-struct _Indirect_value_impl<projected<_It, _Proj>> {
-    using type = invoke_result_t<_Proj&, _Indirect_value_t<_It>>;
+struct _Projected_impl {
+    struct type : _Projected_difference_type_impl<_It>::_Base {

```

Comment:
I intended to use non-_Ugly conventional `type` for the class name, but the name is probably visible as an injected-class-name. So perhaps the name should be made _Ugly...